### PR TITLE
WTB-363 - Connect Admission Service to SQS

### PR DESCRIPTION
**Pull request by rafaelpinabew**

### CODE REVIEW

File: src/api/config/sections/AwsSection.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import { IsBoolean, IsNotEmpty, IsString, IsUrl } from 'class-validator';
+import { DotEnv } from '..';
+
+import { BaseSection } from './BaseSection';
+
+export class AwsSection extends BaseSection {
```
Comment (cemusta) [inline src/api/config/sections/AwsSection.ts:6]: Namingwise, AwsSection is too generic but we are not gonna use other aws services for a while.

File: src/main.ts [hunk 2]
```diff
@@ -2,10 +2,21 @@ import 'module-alias/register';
...
 
 async function start(): Promise<void> {
   const postgresConnection = new PostgresConnection();
+  const messageConsumerService = new MessageConsumerService(
+    config.aws,
+    handleHospitalCreated,
+    handleHospitalDeleted,
+  );
+  if (config.aws.awsSqsEnabled) {
+    messageConsumerService.start();
```
Comment (cemusta) [inline src/main.ts:18]: if it's started stopping it on graful stop would be nice, after closing db connection.

File: src/controller/AdmissionConfigController.ts [hunk 3]
```diff
@@ -79,8 +83,13 @@ const hospitalDeleted: RequestHandler = async (req, res, next) => {
     if (!hospitalId) {
       res.status(404).send();
     }
-    await snsService.handleHospitalDeleted({
-      hospitalId,
+    await handleHospitalDeleted({
+      type: MessageType.hospitalDelete,
+      version: 23,
```
Comment (cemusta) [inline src/controller/AdmissionConfigController.ts:88]: what is version 23?

File: src/business/messages/messageHandler.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    await requestedDocumentRepository.delete(docIds);
+  }
+
+  await admissionConfigRepository.delete(hospitalId);
+}
+
+export {
+  handleHospitalCreated,
+  handleHospitalDeleted,
+  HospitalDeletedMessage,
```
Comment (cemusta) [inline src/business/messages/messageHandler.ts:38]: do we need to re-export?

**Review by cemusta — COMMENTED** [review]
We should delete and add yarn.lock from the repo. 
Since circleci is using npm for operations it would be better to use npm for your local tests but sure you can use yarn locally without commiting lock file.
(we can reverse circleci to yarn, if so we should remove npm lock and commit yarn lock)

File: src/business/messages/messageHandler.ts (hunk 4 above)
Comment (rafaelpinabew) [reply src/business/messages/messageHandler.ts:38]: Good catch, we don't.

File: src/controller/AdmissionConfigController.ts (hunk 3 above)
Comment (rafaelpinabew) [reply src/controller/AdmissionConfigController.ts:88]: This endpoint should be removed after SQS works receiving the messages. When SQS receives the message it should also include a version number, so in this case we would be "mocking" it. But this endpoint was/is only for testing purposes.

File: src/main.ts (hunk 2 above)
Comment (rafaelpinabew) [reply src/main.ts:18]: makes sense, i'll do 👍

File: src/api/config/sections/AwsSection.ts (hunk 1 above)
Comment (rafaelpinabew) [reply src/api/config/sections/AwsSection.ts:6]: Maybe we could name it AwsSqsSection?

File: src/api/config/sections/ServerSection.ts [hunk 5]
```diff
@@ -28,7 +27,7 @@ export class ServerSection extends BaseSection {
   @IsNotEmpty()
   public httpLogging: boolean;
 
-  constructor(config: DotEnv) {
```
Comment (lutzk) [inline src/api/config/sections/ServerSection.ts:31]: why not use the type?
i mean the name sucks `DotEnv`  we can change
but i dont see the point in redefine that type `Record<string, string | undefined>` again and again

Comment (rafaelpinabew) [reply src/api/config/sections/ServerSection.ts]: I added the type back

**Review by cemusta — APPROVED** [review]

**Pull request merged**